uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/filters/date-type.js

```diff
@@ -1,38 +1,35 @@
 define([
-    'tasks-assignments/filters/filters'
-], function (filters) {
+    'tasks-assignments/filters/filters',
+    'common/lodash'
+], function (filters, _) {
     'use strict';
 
     filters.filter('filterByDateType',['$filter', '$rootScope', '$log', function ($filter, $rootScope, $log) {
         $log.debug('Filter: filterBy.dateType');
 
-        return function(inputArr, dateTypeArr) {
-            var filteredArr = [],
-                i = 0,
-                ii,
-                inputArrLen = inputArr.length,
-                dateTypeArrLen = dateTypeArr.length,
-                dateContactList,
-                dateContactListLen;
-
-            if (!inputArrLen || !dateTypeArrLen) {
-                return inputArr;
+        return function(dateContactList, dateTypeList) {
+            var filteredDateContactList = [];
+
+            if (!dateContactList.length || !dateTypeList.length) {
+                return dateContactList;
             }
 
-            for (i; i < inputArrLen; i++) {
-                ii = 0,
-                dateContactList = inputArr[i].dateContactList,
-                dateContactListLen = dateContactList.length;
+            _.forEach(dateContactList, function (dateContact) {
+              var dateContactList = dateContact.dateContactList;
```

Comment:
`dateContactList` is your func parameter, here you are redefining it. Also `var` will remain redefined outside the scope of the function, so it is actually dangerous. Please use different names or do not define `dateContactList` at all.